JDBC/OAuthSampleApp/src/main/java/OAuthSampleApp.java

```diff
@@ -18,44 +18,121 @@
-    private String clientSecret = "";
+	private static void SetUpDbForOAuth() throws Exception
+    {
+		String connectionString = getConnectionString(prop.getProperty("ConnectionString"));
+    	vConnection = DriverManager.getConnection(connectionString);
+		System.out.println("SetUpDbForOAuth");
+    	String CONNECTION_STRING = prop.getProperty("ConnectionString");    
+        String USER = prop.getProperty("User");
+        String CLIENT_ID = prop.getProperty("ClientId");
+        String CLIENT_SECRET = prop.getProperty("ClientSecret");
+        String INTROSPECT_URL = prop.getProperty("TokenUrl") + "introspect";
+       	Statement st = vConnection.createStatement(); 
+       	
+		st.execute("DROP USER IF EXISTS " + USER + " CASCADE;");
+		st.execute("DROP AUTHENTICATION IF EXISTS v_oauth CASCADE;");
+		st.execute("CREATE AUTHENTICATION v_oauth METHOD 'oauth' LOCAL;");//HOST '0.0.0.0/0';");//getOAuthHost());
+		st.execute("ALTER AUTHENTICATION v_oauth SET client_id= '" + CLIENT_ID + "';");
+		st.execute("ALTER AUTHENTICATION v_oauth SET client_secret= '" + CLIENT_SECRET + "';");
+		st.execute("ALTER AUTHENTICATION v_oauth SET introspect_url = '" + INTROSPECT_URL + "';");
+		st.execute("CREATE USER " + USER + ";");
+		st.execute("GRANT AUTHENTICATION v_oauth TO " + USER + ";");
+		st.execute("GRANT ALL ON SCHEMA PUBLIC TO " + USER +";");
+		st.close();
+    }
+    
+    
+    private static void TearDown() throws Exception
+    {
+		System.out.println("TearDown");
+    	Statement st = vConnection.createStatement(); 
+    	String USER = prop.getProperty("User");
+    //    st.execute("DROP USER IF EXISTS " + USER + " CASCADE");
+    //    st.execute("DROP AUTHENTICATION IF EXISTS v_oauth CASCADE");
+        vConnection.close();
+    }
 
-    private static Connection connectToDB(String host, String port, String dbName, String accessToken,
-            String refreshToken, String clientSecret) throws SQLException {
+	private static Connection connectToDB(String accessToken, String clientId, String clientSecret, String tokenUrl) throws SQLException 
+    {
         Properties jdbcOptions = new Properties();
         jdbcOptions.put("oauthaccesstoken", accessToken);
-        jdbcOptions.put("oauthrefreshtoken", refreshToken);
+        jdbcOptions.put("oauthtokenurl", tokenUrl);
+        jdbcOptions.put("oauthclientid", clientId);
         jdbcOptions.put("oauthclientsecret", clientSecret);
 
-        return DriverManager.getConnection(
-                "jdbc:vertica://" + host + ":" + port + "/" + dbName, jdbcOptions);
+		return DriverManager.getConnection(
+                "jdbc:vertica://" + "172.17.0.45" + ":" + "5433" + "/" + "verticadb21451", jdbcOptions);
+    }
+
+    private static void ConnectToDatabase() throws SQLException {
+		System.out.println("ConnectToDatabase");
+		String accessToken = System.getProperty(OAUTH_ACCESS_TOKEN_VAR_STRING);
+
+		if( null == accessToken || accessToken.isEmpty())
```

Comment:
If there's no access token, there needs to be a message or an exception or the user will be confused that no attempt to connect was made.